src/main/java/com/pack/combopack/packer/GreedyPackager.java

```diff
@@ -2,32 +2,47 @@
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
 
+import com.pack.combopack.bean.BinPack;
 import com.pack.combopack.bean.Packable;
-import com.pack.combopack.exception.PackagingException;
-
-public class GreedyPackager extends AbstractKnapSackPackager {
-
-	@Override
-	public <T extends Packable> List<T> pack(List<T> packables, double maxWeight) throws PackagingException{
-
-		//TODO: Need to AOP in future for separation of concern and remove validate  and removeUnwantedItems call
-		validate(packables, maxWeight);
-		packables=removeUnwantedItems(packables, maxWeight);
-		if(packables.isEmpty() )
-			return null;
-		Collections.sort(packables, packables.get(0).profitableComparator());
-		List<T> selectedForPacking = new ArrayList<T>();
-		Double totalSelectItemWeght = 0d;
-		for (T packable : packables) {
-			if ((totalSelectItemWeght + packable.getWeight()) > maxWeight)
-				continue;
-			selectedForPacking.add(packable);
-			totalSelectItemWeght += packable.getWeight();
-		}
-
-		return selectedForPacking;
-	}
+
+public final class GreedyPackager extends AbstractKnapSackPackager {
+
+   final private Comparator<Packable> profitableComparator = (v1, v2) -> Double.compare(v2.getValue() / v2.getWeight(),
+            v1.getValue() / v1.getWeight());
+
+    @Override
+    public List<Packable> pack(BinPack inputPack)  {
+
+        validate(inputPack);
+
+        List<Packable> packables = inputPack.getPackableBins();
+
+        if (packables.isEmpty())
+            return null;
```

Comment:
**Issue**: Returning null for empty packables might lead to potential NullPointerExceptions in consuming code. <br> **Fix**: Return an empty list instead of null to avoid potential NullPointerExceptions. <br> **Code Suggestion**: 
 ```
 -            return null;
 +            return Collections.emptyList();
 ```